arches/app/views/api/__init__.py

```diff
@@ -2383,3 +2383,128 @@ def delete(self, request, identifier=None):
                 status=400,
             )
         return JSONResponse(status=204)
+
+
+class UserPreference(APIBase):
+    def get(self, request, identifier=None):
+        """
+        Returns specific user preference when given uuid
+        If no id specified, and depending on user permissions, returns a list of all user preferences
+        """
+
+        # check user permission
+        overall_access = False
+        if request.user.groups.filter(name="Application Administrator").exists():
+            overall_access = True
+
+        if identifier:
+            try:
+                returned_user_preference = models.UserPreference.objects.get(
+                    pk=identifier
+                )
+            except:
+                return JSONErrorResponse(
+                    _("User Preference GET request failed"),
+                    _("No User Preference with this id"),
+                    status=404,
+                )
+            if overall_access or returned_user_preference.user == request.user:
+                response_data = returned_user_preference
+            else:
+                return JSONErrorResponse(
+                    _("User Preference GET request failed"),
+                    _("You do not have access to view this userpreferenceid"),
+                    status=403,
+                )
+        else:
+            if overall_access:  # give view to everything
+                response_data = [x for x in models.UserPreference.objects.all()]
+            else:
+                response_data = [
+                    x for x in models.UserPreference.objects.filter(user=request.user)
+                ]
+
+        return JSONResponse(response_data)
+
+    @method_decorator(group_required("Application Administrator", raise_exception=True))
+    def post(self, request, identifier=None):
+        if identifier:
+            return JSONErrorResponse(
+                _("User Preference creation failed"),
+                _(
+                    "POST request should not have a userpreferenceid provided in the URL"
+                ),
+                status=400,
+            )
```

Comment:
And if the detail view is the only view implementing a post, then you don't have to check for this error condition.